app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt

```diff
@@ -0,0 +1,124 @@
+package com.github.multimatum_team.multimatum
+
+import android.app.Application
+import android.graphics.Color
+import android.graphics.Typeface
+import android.view.View
+import android.widget.ListView
+import android.widget.TextView
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.multimatum_team.multimatum.model.Deadline
+import com.github.multimatum_team.multimatum.model.DeadlineAdapter
+import com.github.multimatum_team.multimatum.model.DeadlineState
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.time.LocalDate
+
+@RunWith(AndroidJUnit4::class)
+class DeadlineAdapterTest {
+
+    private lateinit var adapter: DeadlineAdapter
+    private var context: Application? = null
+    private lateinit var list: List<Deadline>
+
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        list = listOf(Deadline("Number 1",DeadlineState.TODO, LocalDate.now().plusDays(1)),
+            Deadline("Number 2",DeadlineState.TODO, LocalDate.now().plusDays(7)),
+            Deadline("Number 3",DeadlineState.DONE, LocalDate.of(2022, 3,30)),
+            Deadline("Number 4",DeadlineState.TODO, LocalDate.of(2022, 3,1)))
+        context = ApplicationProvider.getApplicationContext()
+        adapter = DeadlineAdapter(context!!, list)
+    }
+
+
+    @Test
+    fun `GetCount should give the correct count`(){
+        Assert.assertEquals(adapter.count, 4)
+    }
+
+    @Test
+    fun `GetItem should give the correct deadline in the list`(){
+        Assert.assertEquals(adapter.getItem(0), list[0])
+        Assert.assertEquals(adapter.getItem(1), list[1])
+        Assert.assertEquals(adapter.getItem(2), list[2])
+        Assert.assertEquals(adapter.getItem(3), list[3])
+    }
+
+    @Test
+    fun `GetItemId should give the correct Id of an item`(){
+        Assert.assertEquals(adapter.getItemId(0), 0.toLong())
+        Assert.assertEquals(adapter.getItemId(1), 1.toLong())
+        Assert.assertEquals(adapter.getItemId(2), 2.toLong())
+        Assert.assertEquals(adapter.getItemId(3), 3.toLong())
+    }
+
+    @Test
+    fun `GetView should show properly the first deadline who is due in 1 day in red`(){
+        val parent = ListView(context)
+        val listItemView: View = adapter.getView(0,null,parent)
+        //check title
+        Assert.assertEquals("Number 1",(listItemView.findViewById(R.id.deadline_list_title) as TextView).text)
```

Comment:
Same as before, you can pass a type parameter to `findViewById` instead of casting its return value